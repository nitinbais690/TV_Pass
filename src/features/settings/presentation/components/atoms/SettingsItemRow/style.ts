import { scale, selectDeviceType } from 'qp-common-ui';
import { appFonts, appFontStyle, appPaddingValues } from 'core/styles/AppStyles';
import { appFlexStyles } from 'core/styles/FlexStyles';
import { StyleSheet } from 'react-native';

export function SettingsMenuRowStyles(appColors: any) {
    return StyleSheet.create({
        container: {
            ...appFlexStyles.flexColumn,
        },
        divider: {
            width: '100%',
            borderBottomWidth: scale(1, 0),
            opacity: 0.07,
            borderColor: appColors.secondary,
        },
        titleRowContainer: {
            ...appFlexStyles.rowVerticalAlignCenter,
            paddingTop: selectDeviceType({ Handset: scale(19, 0) }, scale(32, 0)),
        },
        titleText: {
            color: appColors.secondary,
            paddingLeft: appPaddingValues.sm,
            ...appFontStyle.body2,
            fontFamily: appFonts.semibold,
        },
        subTitleText: {
            color: appColors.secondary,
            ...appFontStyle.body3,
            fontFamily: appFonts.semibold,
            paddingTop: scale(9, 0),
            paddingBottom: selectDeviceType({ Handset: scale(16, 0) }, scale(26, 0)),
            opacity: 0.3,
        },
        arrowIcon: {
            marginStart: 'auto',
            width: selectDeviceType({ Handset: scale(24, 0) }, scale(34, 0)),
            height: selectDeviceType({ Handset: scale(24, 0) }, scale(34, 0)),
        },
    });
}
