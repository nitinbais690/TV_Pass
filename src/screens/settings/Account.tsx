import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { useAppPreferencesState } from 'utils/AppPreferencesContext';
import { appFonts, tvPixelSizeForLayout } from '../../../AppStyles';
import { useLocalization } from 'contexts/LocalizationContext';
import AccountIcon from '../../../assets/images/account_tv.svg';

/**
 * A component to display the settings/Account screen.
 */
const Account = (): JSX.Element => {
    const prefs = useAppPreferencesState();
    const { appTheme } = prefs;
    const { strings } = useLocalization();
    let { appColors } = appTheme && appTheme(prefs);

    const defaultStyles = StyleSheet.create({
        creditIcon: { opacity: 0.5, paddingVertical: tvPixelSizeForLayout(10) },
        subscriptionHeading: {
            paddingTop: tvPixelSizeForLayout(40),
            color: appColors.secondary,
            fontSize: tvPixelSizeForLayout(75),
            fontFamily: appFonts.primary,
            fontWeight: '600',
        },
        subscriptionDescription: {
            paddingTop: tvPixelSizeForLayout(40),
            color: appColors.tertiary,
            fontSize: tvPixelSizeForLayout(32),
            fontFamily: appFonts.primary,
        },
    });

    return (
        <>
            <AccountIcon width={tvPixelSizeForLayout(120)} height={tvPixelSizeForLayout(77)} />
            <Text style={defaultStyles.subscriptionHeading}>{strings['Account.SubscriptionHeadingTv']}</Text>
            <Text style={defaultStyles.subscriptionDescription}>{strings['Account.SubscriptionDescriptionTV']}</Text>
        </>
    );
};

export default Account;
