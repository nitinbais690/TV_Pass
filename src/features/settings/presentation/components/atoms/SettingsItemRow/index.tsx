import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { SettingsMenuRowStyles } from './style';
import RightArrowIcon from 'assets/images/right_arrow_icon.svg';
import useAppColors from 'core/presentation/hooks/use-app-colors';

const SettingsItemRow = (props: SettingsItemRowProps) => {
    const styles = SettingsMenuRowStyles(useAppColors());

    return (
        <TouchableOpacity onPress={props.onPress}>
            <View style={styles.container}>
                <View style={styles.divider} />
                <View style={styles.titleRowContainer}>
                    {props.image}
                    <Text style={styles.titleText}>{props.title}</Text>
                    <RightArrowIcon style={styles.arrowIcon} />
                </View>
                <Text style={styles.subTitleText}>{props.subtitle}</Text>
            </View>
        </TouchableOpacity>
    );
};

interface SettingsItemRowProps {
    image: JSX.Element | undefined;
    title: string;
    subtitle: string;
    onPress?: () => void;
}

export default SettingsItemRow;
