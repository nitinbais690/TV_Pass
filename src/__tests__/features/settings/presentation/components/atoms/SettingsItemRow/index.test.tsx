import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import SettingsItemRow from 'features/settings/presentation/components/atoms/SettingsItemRow';
import Account from 'assets/images/account_outline.svg';

jest.mock('core/presentation/hooks/use-app-colors');

describe('settingsItemRow', () => {
    const title = 'User profiles';
    const subtitle = 'Manage your profiles';
    const image = <Account />;

    it('renders correctly', () => {
        const { container, getByText } = render(<SettingsItemRow title={title} subtitle={subtitle} image={image} />);
        expect(getByText(title)).toBeDefined();
        expect(getByText(subtitle)).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });
});
