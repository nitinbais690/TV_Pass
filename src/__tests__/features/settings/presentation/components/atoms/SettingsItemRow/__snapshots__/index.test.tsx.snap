// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settingsItemRow renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#FFFFFF",
            "opacity": 0.07,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingTop": 32,
          }
        }
      >
        <SvgMock />
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "ProximaNova-Semibold",
              "fontSize": 20,
              "fontWeight": "600",
              "paddingLeft": 20,
            }
          }
        >
          User profiles
        </Text>
        <SvgMock
          style={
            Object {
              "height": 34,
              "marginStart": "auto",
              "width": 34,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Semibold",
            "fontSize": 18,
            "fontWeight": "600",
            "opacity": 0.3,
            "paddingBottom": 26,
            "paddingTop": 9,
          }
        }
      >
        Manage your profiles
      </Text>
    </View>
  </View>
</View>
`;
